app/utils/transactionSerialization.ts

```diff
@@ -34,28 +34,42 @@ import {
     serializeList,
     serializeCredentialDeploymentInformation,
     serializeBoolean,
+    encodeWord16,
 } from './serializationHelpers';
+import { encodeAsCBOR } from './cborHelper';
+
+function serializeMemo(memo: string) {
+    const encoded = encodeAsCBOR(memo);
+    const length = encodeWord16(encoded.length);
+    return Buffer.concat([length, encoded]);
+}
 
 function serializeSimpleTransfer(payload: SimpleTransferPayload) {
-    const size = 1 + 32 + 8;
-    const serialized = new Uint8Array(size);
+    const kind = putInt8(TransactionKind.Simple_transfer);
+    const address = base58ToBuffer(payload.toAddress);
+    const amount = encodeWord64(BigInt(payload.amount));
+    return Buffer.concat([kind, address, amount]);
+}
 
-    serialized[0] = TransactionKind.Simple_transfer;
-    putBase58Check(serialized, 1, payload.toAddress);
-    put(serialized, 32 + 1, encodeWord64(BigInt(payload.amount)));
-    return Buffer.from(serialized);
+function serializeSimpleTransferWithMemo(payload: SimpleTransferPayload) {
+    if (!payload.memo) {
```

Comment:
It should not be necessary to have this check here if we type the input parameter better.